src/Converter/LineItemsConverter.php

```diff
@@ -18,44 +18,119 @@
 use Sylius\Component\Core\Model\AdjustmentInterface;
 use Sylius\Component\Core\Model\OrderInterface;
 use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\Component\Order\Model\AdjustableInterface;
 use Sylius\InvoicingPlugin\Entity\LineItem;
+use Sylius\InvoicingPlugin\Exception\MoreThanOneTaxAdjustment;
+use Sylius\InvoicingPlugin\Provider\TaxRateProviderInterface;
+use Webmozart\Assert\Assert;
 
 final class LineItemsConverter implements LineItemsConverterInterface
 {
+    /** @var TaxRateProviderInterface */
+    private $taxRateProvider;
+
+    public function __construct(TaxRateProviderInterface $taxRateProvider)
+    {
+        $this->taxRateProvider = $taxRateProvider;
+    }
+
     public function convert(OrderInterface $order): Collection
     {
         $orderItems = $order->getItems();
-        $shippingAdjustments = $order->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT);
+
         $lineItems = new ArrayCollection();
 
+        $this->addLineItemForShipment($lineItems, $order);
+
         /** @var OrderItemInterface $orderItem */
         foreach ($orderItems as $orderItem) {
             $variant = $orderItem->getVariant();
 
             $lineItems->add(new LineItem(
                 $orderItem->getProductName(),
                 $orderItem->getQuantity(),
-                $orderItem->getUnitPrice(),
-                $orderItem->getSubtotal(),
+                (int) ($orderItem->getTotal() / $orderItem->getQuantity() - $orderItem->getTaxTotal() / $orderItem->getQuantity()),
```

Comment:
But I'm not sure if we will end up with a proper amount, as not every unit has the same value 🤔 
```suggestion
                (int) ($orderItem->getTotal() - $orderItem->getTaxTotal()) / $orderItem->getQuantity(),
```

Reply:
And this one case is also defined in task, if the units have different values, they should be split into different line items